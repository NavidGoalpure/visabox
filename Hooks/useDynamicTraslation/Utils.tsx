import {
  Languages,
  MultiLanguageText,
  MultiLanguageTextArray,
} from 'Interfaces';
import { useLocale } from 'Hooks/useLocale';

/**
 *  یک گزاره را گرفته،‌بسته به زبان کاربر که از یوارال فهمیده میشود، ترجمه مناسب را برمیگرداند
 * @param  statementObj عبارتی که از سرور میاد، در زبان های مختلف
 * @return      جمله نهایی به زبان کاربر- که از یوآرال گرفته شده
 */
export const translatedObject = ({
  statementObj,
}: {
  statementObj: MultiLanguageText | undefined;
}): string => {
  const { locale } = useLocale();

  if (!statementObj || !locale) return '';

  if (
    typeof statementObj[locale] !== 'undefined' &&
    statementObj[locale] !== ''
  )
    return statementObj[locale] || statementObj[Languages.en];

  return statementObj[Languages.en];

  //
};
///////////////////////
export const translateDynamicArray = (
  statementObj: MultiLanguageTextArray | undefined
): string[] => {
  if (!statementObj) return [];
  const { locale } = useLocale();
  if (!statementObj || !locale) return [];

  if (typeof statementObj[locale] !== 'undefined')
    return (
      statementObj[locale] ||
      //defensive
      statementObj[Languages.en]
    );

  return statementObj[Languages.en];
};
