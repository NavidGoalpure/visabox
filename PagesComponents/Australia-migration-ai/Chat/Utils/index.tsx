import { getClientDetail } from 'Queries/client';
import { LOOKUP_STORE } from '../const';
import { ILookupEnum } from '../Interface';

export function getLookupLabel(lookup: ILookupEnum) {
  return LOOKUP_STORE?.[lookup];
}

export function convertMarkdownToHTML(markdown: string): string {
  // Handle citations: <Citation title="foo" href="bar"/>
  const citationRegex = /<Citation title="([^"]+)" href="([^"]+)"\s*\/>/g;
  markdown = markdown.replace(citationRegex, '<a href="$2">$1</a>');

  // Handle links: [text](url)
  const linkRegex = /\[([^\]]+)\]\(([^)]+)\)/g;
  markdown = markdown.replace(linkRegex, '<a href="$2">$1</a>');

  // Handle bold: **text**
  const boldRegex = /\*\*(.*?)\*\*/g;
  markdown = markdown.replace(boldRegex, '<strong>$1</strong>');

  // Handle italic: _text_
  const italicRegex = /_(.*?)_/g;
  markdown = markdown.replace(italicRegex, '<em>$1</em>');

  // Handle headers: # Header 1, ## Header 2, ... ###### Header 6
  for (let i = 6; i >= 1; i--) {
    const headerRegex = new RegExp(`^#{${i}}\\s+(.*)$`, 'gm');
    markdown = markdown.replace(
      headerRegex,
      (_, text) => `<h${i}>${text}</h${i}>`
    );
  }

  // Handle list items: * or -
  const listItemRegex = /(?:\*|\-)\s+(.*)/g;
  markdown = markdown.replace(listItemRegex, '<li>$1</li>');

  // Handle unordered list: * list item
  const unorderedListRegex = /(?:^\*.*\n)+/gm;
  markdown = markdown.replace(unorderedListRegex, '<ul>$&</ul>');

  // Handle ordered list: 1. list item
  const orderedListRegex = /(?:^\d+\..*\n)+/gm;
  markdown = markdown.replace(orderedListRegex, '<ol>$&</ol>');

  // Handle blockquote: > Blockquote
  const blockquoteRegex = /^>(.*)$/gm;
  markdown = markdown.replace(blockquoteRegex, '<blockquote>$1</blockquote>');

  // Handle UI (Underline): __text__
  const uiRegex = /__(.*?)__/g;
  markdown = markdown.replace(uiRegex, '<u>$1</u>');

  // Handle tables: | Header 1 | Header 2 | ... |
  const tableRegex = /^\|(.+)\|$/gm;
  markdown = markdown.replace(tableRegex, (tableRow) => {
    const cells = tableRow
      .split('|')
      .map((cell) => cell.trim())
      .filter(Boolean);
    const isHeader = tableRow.startsWith('|:') && tableRow.endsWith(':|');
    if (isHeader) {
      return (
        '<thead><tr>' +
        cells.map((cell) => `<th>${cell}</th>`).join('') +
        '</tr></thead><tbody>'
      );
    }
    return '<tr>' + cells.map((cell) => `<td>${cell}</td>`).join('') + '</tr>';
  });
  // Close the tbody for table if opened
  markdown = markdown.replace(/<tbody>/g, '</tbody>');

  return markdown;
}
