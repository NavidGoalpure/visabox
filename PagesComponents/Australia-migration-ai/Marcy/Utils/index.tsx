import { LOOKUP_STORE } from '../const';
import { ILookupEnum } from '../Interface';

export function getLookupLabel(lookup: ILookupEnum) {
  return LOOKUP_STORE?.[lookup];
}
