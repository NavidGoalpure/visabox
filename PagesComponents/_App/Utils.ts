import { hotjar } from 'react-hotjar';

/**
 * این فانکشن تصمیم میگیره که هاتجر رو در یک صفحه خاص فعال کنیم یا خیر
 * اگه خواستیم صحفه ای فعال باشه اون رو در لیست اکیتوها میزاریم
 * اگه خواستیم صحفه ای رو حذف کنیم اون رو در لیست دی اکتیوها میزاریم
 * همچنین میتونیم لیست اکیتوها رو «آل» بزاریم و فقط یک صفحات خاصی رو دی اکتیو کنیم
 * @export
 * @param {string} currentRoute
 * @return {*}  {boolean}
 */
export function doMustActiveHotjar(currentRoute: string): boolean {
  // روت هایی که میخوایم هاتجر در اونها فعال باشه
  // const MUST_ACTIVE_ROUTES = ['/agents/forms-wall', '/clients/point-calculator'];
  const MUST_ACTIVE_ROUTES = ['ALL'];

  //روت هایی که نمیخوایم هاتجر در اونها فعال باشه
  const MUST_DEACTIVE_ROUTES = ['/404'];

  ////////////result///////////////
  // اگه تو لیست دی اکتیوها بود فالس برگردون
  if (MUST_DEACTIVE_ROUTES.includes(currentRoute)) return false;

  // اگه قرار شد همه صفحات رو اکتیو کنیم ترو برگردون
  //@ts-ignore
  if (MUST_ACTIVE_ROUTES.toString() === ['ALL'].toString()) return true;

  //اگه تو لیست اکتیوها بود ترو برگردون
  return MUST_ACTIVE_ROUTES.includes(currentRoute);
}

/**
 *  این فانکشن برای فعال کردن هاتجر استفاده می‌شود
 *
 * @export
 * @param {string} currentRoute
 */
export function smartActiveHotjar(currentRoute: string) {
  const isActiveHotjar = doMustActiveHotjar(currentRoute);
  console.log('navid isHotjar=', isActiveHotjar);
  if (isActiveHotjar)
    hotjar.initialize(
      Number(process.env.NEXT_PUBLIC_HJID),
      Number(process.env.NEXT_PUBLIC_HJSV)
    );
}
