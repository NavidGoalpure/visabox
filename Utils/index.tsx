import { LocalStorageKeys, ThemeModes } from 'Interfaces';

const getThemeFromLocalStorage = (): ThemeModes => {
  const DEFAULT_THEME = ThemeModes.DARK;
  const res =
    typeof window === 'undefined'
      ? DEFAULT_THEME
      : (localStorage.getItem(LocalStorageKeys.Theme) as ThemeModes);
  return res;
};
//

//
function isItOnLive(): boolean {
  const smartLocation =
    typeof window === 'undefined' ? '' : window?.location?.href;
  if (
    smartLocation.includes('https://marabox.com') ||
    smartLocation.includes('http://marabox.com') ||
    smartLocation.includes('https://www.marabox.com') ||
    smartLocation.includes('http://www.marabox.com')
  )
    return true;
  return false;
}
function setLocalStorage({
  key,
  value,
  isReloadPage = false,
}: {
  key: LocalStorageKeys;
  value: string;
  isReloadPage?: boolean;
}) {
  if (typeof window === 'undefined') return;
  //
  window.localStorage.setItem(key, value);
  //
  if (isReloadPage) window.location.reload();
}
function getLocalStorage(key: LocalStorageKeys): string | null {
  if (typeof window === 'undefined') return null;
  //
  return window.localStorage.getItem(key);
  //
}
const copyContent = async (text: string) => {
  try {
    await navigator.clipboard.writeText(text);
  } catch (err) {
    console.error('Failed to copy: ', err);
    //navid implement toast
  }
};
export {
  getThemeFromLocalStorage,
  isItOnLive,
  setLocalStorage,
  getLocalStorage,
  copyContent,
};
