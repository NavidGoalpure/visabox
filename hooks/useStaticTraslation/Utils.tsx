import { MultiLanguageText } from 'interfaces';
import { useLocale } from 'hooks/useLocale';

/**
 *  بوسیله این فانکشن که برای متون هوشمند استفاده میشود میتوان کلماتی از متن را که به صورت تمپلیت مشخص شده اند
 * با کلمات دیگری جایگزین نمود
 * @param  localeSentence متن که ممکن است به هر زبانی باشد
 * @param  aliases جفت کلیدهایی که با آن مشخص میکنیم چه کلماتی باید با چه کلماتی جایگزین شوند
 * @return       جمله نهایی که تمام کلمات تمپلیت آن با کلمات مقصد جایگزین شده اند
 */
function convertAllTempKeysWithAllAliases({
  localeSentence,
  aliases,
}: {
  localeSentence: string;
  aliases: Record<string, string>[];
}): string {
  let mylocaleSentence = localeSentence;
  aliases.map((pair) => {
    console.log('navid mylocaleSentence=', mylocaleSentence);
    console.log('navid pair=', pair);

    mylocaleSentence = convertKeyWithValueInString({
      senttence: mylocaleSentence,
      pair,
    });
  });
  return mylocaleSentence;
}

////////////////
/**
 *  یک استرینگ و یک آبجکت را گرفته، کلید آبجکت را با ولیوی آن -در استرینگ- جایگزین میکند
 * @param  senttence جمله
 * @param  pair آبجکت
 * @return      استرینگ جایگزین شده
 */
function convertKeyWithValueInString({
  senttence,
  pair,
}: {
  senttence: string;
  pair: Record<string, string>;
}): string {
  return senttence.replace(`${Object.keys(pair)[0]}`, Object.values(pair)[0]);
}

////////////////
/**
 *  یک گزاره را گرفته،‌بسته به زبان کاربر که از یوارال فهمیده میشود، ترجمه مناسب را برمیگرداند
 * @param  statementKey جمله
 * @param  statements آبجکتی از تمام گزاره های یک پیج یا کامپوننت به زبان های مختلف
 * @param  aliases جفت کلیدهایی که با آن مشخص میکنیم چه کلماتی باید با چه کلماتی جایگزین شوند
 * @return      جمله نهایی به زبان کاربر- که از یوآرال گرفته شده- بعد از جایگزینی تمپلیت ها با آلیاس ها
 */
export const translatedObject = ({
  statementKey,
  statements,
  aliases,
}: {
  statementKey: string;
  statements: Record<string, MultiLanguageText>;
  aliases?: Record<string, string>[];
}): string => {
  const { locale } = useLocale();
  if (!statementKey || !locale) return '';

  if (!aliases || aliases.length === 0)
    return statements?.[statementKey]?.[locale] || '';
  //

  return convertAllTempKeysWithAllAliases({
    localeSentence: statements?.[statementKey]?.[locale] || '',
    aliases,
  });
};
