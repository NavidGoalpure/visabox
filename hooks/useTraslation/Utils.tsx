import { useRouter } from 'next/router';
import { MultiLanguageText } from 'interfaces';
import { proxyUrlLocaleToLocale } from 'Utils/Language';

/**
 *  بوسیله این فانکشن که برای متون هوشمند استفاده میشود میتوان کلماتی از متن را که به صورت تمپلیت مشخص شده اند
 * با کلمات دیگری جایگزین نمود
 * @param  localeSentence متن که ممکن است به هر زبانی باشد
 * @param  aliases جفت کلیدهایی که با آن مشخص میکنیم چه کلماتی باید با چه کلماتی جایگزین شوند
 * @return       جمله نهایی که تمام کلمات تمپلیت آن با کلمات مقصد جایگزین شده اند
 */
function convertAllTempKeysWithAllAliases({
  localeSentence,
  aliases,
}: {
  localeSentence: string;
  aliases: Record<string, string>[];
}): string {
  let mylocaleSentence = localeSentence;
  aliases.map((pair) => {
    mylocaleSentence = convertKeyWithValueInString({
      senttence: mylocaleSentence,
      pair,
    });
  });
  return mylocaleSentence;
}

////////////////
/**
 *  یک استرینگ و یک آبجکت را گرفته، کلید آبجکت را با ولیوی آن -در استرینگ- جایگزین میکند
 * @param  senttence جمله
 * @param  pair آبجکت
 * @return      استرینگ جایگزین شده
 */
function convertKeyWithValueInString({
  senttence,
  pair,
}: {
  senttence: string;
  pair: Record<string, string>;
}): string {
  return senttence.replace(
    `{{${Object.keys(pair)[0]}}}`,
    Object.values(pair)[0]
  );
}

////////////////
/**
 *  یک گزاره را گرفته،‌بسته به زبان کاربر که از یوارال فهمیده میشود، ترجمه مناسب را برمیگرداند
 * @param  statementKey جمله
 * @param  statements آبجکتی از تمام گزاره های یک پیج یا کامپوننت به زبان های مختلف
 * @param  aliases جفت کلیدهایی که با آن مشخص میکنیم چه کلماتی باید با چه کلماتی جایگزین شوند
 * @return      جمله نهایی به زبان کاربر- که از یوآرال گرفته شده- بعد از جایگزینی تمپلیت ها با آلیاس ها
 */
export const translatedObject = ({
  statementKey,
  statements,
  aliases,
}: {
  statementKey: string;
  statements: Record<string, MultiLanguageText>;
  aliases?: Record<string, string>[];
}): string => {
  const {
    query: { locale },
  } = useRouter();

  if (!statementKey || !locale) return '';
  const smartLocale = proxyUrlLocaleToLocale(locale);

  if (!aliases || aliases.length === 0)
    return statements?.[statementKey]?.[smartLocale] || '';
  //
  return convertAllTempKeysWithAllAliases({
    localeSentence: statements?.[statementKey]?.[smartLocale] || '',
    aliases,
  });
};
