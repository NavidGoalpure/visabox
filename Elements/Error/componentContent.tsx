import { globalComponentStatements, LanguageKeys } from 'Consts/language';
import { useStaticTranslation } from 'Hooks/useStaticTraslation';

import React, { ReactElement } from 'react';
import styled from 'styled-components/macro';
import { componentTitleStyle } from 'Styles/Theme/Component';
import { MdError } from 'react-icons/md';


export const ComponentError: React.FC = () => {
  const { t } = useStaticTranslation(globalComponentStatements);
    return (
      <Container>
        <ErrorIcon />
        <p>{t(LanguageKeys.SomethingWentWrong)}</p>
      </Container>
    );
};
const ErrorIcon = styled(MdError)`
  color: var(--color-fail1);
  width: 3rem;
  height: 3rem;
  margin-inline-end: 1rem;
`;

const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  p {
    ${componentTitleStyle}
  }
`;
