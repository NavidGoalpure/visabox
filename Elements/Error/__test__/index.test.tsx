import { screen, render, fireEvent } from 'Utils/test-utils';
import { ComponentError } from '../componentContent';
import mockRouter from 'next-router-mock';
import renderer from 'react-test-renderer';
jest.mock('next/router', () => require('next-router-mock'));

describe('ComponentError', () => {
  /////snapshot///////
  it('renders ComponentError correctly ', () => {
    const openComponentError = renderer.create(<ComponentError />).toJSON();
    expect(openComponentError).toMatchSnapshot();
  });
});
