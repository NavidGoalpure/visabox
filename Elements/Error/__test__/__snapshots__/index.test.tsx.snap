// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentError renders ComponentError correctly  1`] = `
<div
  className="componentContent__Container-sc-qkubvc-1 eBYLNH"
>
  <svg
    className="componentContent__ErrorIcon-sc-qkubvc-0 ixAFon"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
    />
  </svg>
  <h3 />
  <p />
</div>
`;

exports[`ComponentError renders open accordin correctly  1`] = `
<div
  className="componentContent__Container-sc-qkubvc-1 eBYLNH"
>
  <svg
    className="componentContent__ErrorIcon-sc-qkubvc-0 ixAFon"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
    />
  </svg>
  <h3 />
  <p />
</div>
`;
